### PR TITLE
Notifications in favicon

### PR DESCRIPTION
**Pull request by Paparoni**
Functions for adding notifications in favicon,
Example here: http://heark-test-notif.bitballoon.com/

### CODE REVIEW

**Comment by danimesq** [comment]
Thank you :smile:

**Comment by coyotte508** [comment]
Uh no...

I don't get why the `Object.watch`. We have similar stuff (observable) already.

And it's not actually updated?

Edit: Maybe an animation for the first one, but for subsequent ones update the number directly.

**Comment by DaniellMesquita** [comment]
Title change suggest: `Notifications in favicon (#115)`

**Comment by Paparoni** [comment]
It adds an animation for each of them on my end?
And I did not know about observable.. I added `Object.watch` to detect when `notif.count` increased.

**Comment by coyotte508** [comment]
I mean, only one animation is my sugesstion. The animation is too slow for doing one every time.

Also, about the watch thing, better add it only if needed - and it'll be known when implementing.

**Comment by Paparoni** [comment]
Alrighty, I'll do that right now.

**Comment by DaniellMesquita** [comment]
The notification system can be increased.
- [ ] Notification badge counting lines in pm tab
- [ ] Notification badge counting mentions in channel tab
- [ ] Notification badge counting mentions and other notifications in battle tab

If 10 PM messages, 2 mentions and 30 battle tab notifications, all this can be shortened in favicon.

If 10 messages but 5 is from User A and other 5 is from User B, this is counted as 2 in favicon (only counting user messages, not lines).

If 2 mentions, only count in 1 favicon for each channel that this is happened.

If 30 notifications in only one battle tab, count as 1. Increase regarding number of channels in which this event repeats.

**Comment by Paparoni** [comment]
That should do it

File: app/assets/javascript/notif.js [hunk 1]
```diff
@@ -10,67 +10,47 @@ notif.dec() - Removes 1 notification
...
+        } else if (notif.count == 0) {
+            notif.faviconNoAnim.badge("");
+        } else {
+            notif.faviconNoAnim.badge(notif.count)
         }
-    };
+
+    },
+    clear: function() {
+        notif.count = 0
```
Comment (coyotte508) [inline app/assets/javascript/notif.js:45]: add notif.one_count = 0 here?

**Pull request closed without merging**